src/pages/iou/request/step/IOURequestStepWaypoint.js

```diff
@@ -0,0 +1,273 @@
+        Navigation.goBack(ROUTES.MONEY_REQUEST_DISTANCE_TAB.getRoute(iouType));
+    };
+
+    /**
+     * @param {Object} values
+     * @param {String} values.lat
+     * @param {String} values.lng
+     * @param {String} values.address
+     */
+    const selectWaypoint = (values) => {
+        const waypoint = {
+            lat: values.lat,
+            lng: values.lng,
+            address: values.address,
+        };
+        Transaction.saveWaypoint(transactionID, pageIndex, waypoint, false);
+        Navigation.goBack(ROUTES.MONEYTEMPORARYFORREFACTOR_REQUEST_CREATE_TAB_DISTANCE.getRoute(iouType, transactionID, reportID));
+    };
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            onEntryTransitionEnd={() => textInput.current && textInput.current.focus()}
+            shouldEnableMaxHeight
+            testID={IOURequestStepWaypoint.displayName}
+        >
+            <FullPageNotFoundView shouldShow={(Number.isNaN(parsedWaypointIndex) || parsedWaypointIndex < 0 || parsedWaypointIndex > waypointCount) && isFocused}>
+                <HeaderWithBackButton
+                    title={translate(wayPointDescriptionKey)}
+                    shouldShowBackButton
+                    onBackButtonPress={() => {
+                        Navigation.goBack(ROUTES.MONEY_REQUEST_DISTANCE_TAB.getRoute(iouType));
+                    }}
+                    shouldShowThreeDotsButton={shouldShowThreeDotsButton}
+                    threeDotsAnchorPosition={styles.threeDotsPopoverOffset(windowWidth)}
+                    threeDotsMenuItems={[
+                        {
+                            icon: Expensicons.Trashcan,
+                            text: translate('distance.deleteWaypoint'),
+                            onSelected: () => setIsDeleteStopModalOpen(true),
+                        },
+                    ]}
+                />
+                <ConfirmModal
+                    title={translate('distance.deleteWaypoint')}
+                    isVisible={isDeleteStopModalOpen}
+                    onConfirm={deleteStopAndHideModal}
+                    onCancel={() => setIsDeleteStopModalOpen(false)}
+                    prompt={translate('distance.deleteWaypointConfirmation')}
+                    confirmText={translate('common.delete')}
+                    cancelText={translate('common.cancel')}
+                    danger
+                />
+                <FormProvider
+                    style={[styles.flexGrow1, styles.mh5]}
+                    formID={ONYXKEYS.FORMS.WAYPOINT_FORM}
+                    enabledWhenOffline
+                    validate={validate}
+                    onSubmit={submit}
+                    shouldValidateOnChange={false}
+                    shouldValidateOnBlur={false}
+                    submitButtonText={translate('common.save')}
+                >
+                    <View>
+                        <AddressSearch
```

Comment:
I think we need to wrap this in `InputWrapperWithRef` for it to work, like we do for other inputs, [example](https://github.com/Expensify/App/blob/e2c1c031d388b6e7277249d87d3e13197f2cbf3a/src/pages/iou/MoneyRequestDescriptionPage.js#L127-L146)